engine/storage/image/src/main/java/org/apache/cloudstack/storage/image/TemplateServiceImpl.java

```diff
@@ -1052,6 +1052,9 @@ public void syncTemplateToRegionStore(long templateId, DataStore store) {
         DataObject templateOnStore = destStore.create(tmplForCopy);
         templateOnStore.processEvent(Event.CreateOnlyRequested);
 
+        // I am the big assumer^Wpretender
```

Comment:
@shwstppr use instanceof to check; avoid exceptions

Reply:
Ok, i checked the code path and it will be an `TemplateObject` always, but you are right.

Reply:
I changed the code but the net effect stays the same. If it isn't a `TemplateObject` an exception is thrown and should be thrown. we'll be in the wrong spot in that case.
@rhtyd ?